examples/pytorch/ogb/ngnn/main.py

```diff
@@ -0,0 +1,437 @@
+def test(model, predictor, g, valg, x, split_edge, evaluator, batch_size):
+    model.eval()
+    predictor.eval()
+
+    h = model(g, x)
+
+    pos_train_edge = split_edge['eval_train']['edge'].to(h.device)
+    pos_valid_edge = split_edge['valid']['edge'].to(h.device)
+    neg_valid_edge = split_edge['valid']['edge_neg'].to(h.device)
+    pos_test_edge = split_edge['test']['edge'].to(h.device)
+    neg_test_edge = split_edge['test']['edge_neg'].to(h.device)
+
+    def get_pred(test_edges, h):
+        preds = []
+        for perm in DataLoader(range(test_edges.size(0)), batch_size):
+            edge = test_edges[perm].t()
+            preds += [predictor(h[edge[0]], h[edge[1]]).squeeze().cpu()]
+        pred = torch.cat(preds, dim=0)
+        return pred
+    
+    pos_train_pred = get_pred(pos_train_edge, h)
+    pos_valid_pred = get_pred(pos_valid_edge, h)
+    neg_valid_pred = get_pred(neg_valid_edge, h)
+
+    h = model(valg, x)
+
+    pos_test_pred = get_pred(pos_test_edge, h)
+    neg_test_pred = get_pred(neg_test_edge, h)
+
+    results = {}
+    for K in [20, 50, 100]:
+        evaluator.K = K
+        train_hits = evaluator.eval({
+            'y_pred_pos': pos_train_pred,
+            'y_pred_neg': neg_valid_pred,
+        })[f'hits@{K}']
+        valid_hits = evaluator.eval({
+            'y_pred_pos': pos_valid_pred,
+            'y_pred_neg': neg_valid_pred,
+        })[f'hits@{K}']
+        test_hits = evaluator.eval({
+            'y_pred_pos': pos_test_pred,
+            'y_pred_neg': neg_test_pred,
+        })[f'hits@{K}']
+
+        results[f'Hits@{K}'] = (train_hits, valid_hits, test_hits)
+
+    return results
+
+
+def main():
+    parser = argparse.ArgumentParser(description='OGBL(Full Batch GCN/GraphSage + NGNN)')
+
+    # dataset setting
+    parser.add_argument('--dataset', type=str, default='ogbl-ddi', choices=['ogbl-ddi', 'ogbl-collab', 'ogbl-ppa'])
+
+    # device setting
+    parser.add_argument('--device', type=int, default=0, help='GPU device ID. Use -1 for CPU training.')
+
+    # model structure settings
+    parser.add_argument('--use_sage', action='store_true', help='If not set, use GCN by default.')
+    parser.add_argument('--ngnn_type', type=str, default="none", choices=['none', 'input', 'hidden', 'output', 'all'], help="You can set this value from 'none', 'input', 'hidden' or 'all' to apply NGNN to different GNN layers.")
+    parser.add_argument('--num_layers', type=int, default=3, help='number of GNN layers')
+    parser.add_argument('--hidden_channels', type=int, default=256)
+    parser.add_argument('--dropout', type=float, default=0.0)
+    parser.add_argument('--batch_size', type=int, default=64 * 1024)
+    parser.add_argument('--lr', type=float, default=0.001)
+    parser.add_argument('--epochs', type=int, default=400)
+
+    # training settings
+    parser.add_argument('--use_valedges_as_input', action='store_true', help='Use training + validation edges for inference on test set.')
```

Comment:
It seems that you do not use this argument in `README.md`. Then you can remove it and the related code.

Reply:
OGB allows to use validation edges (together with train edges) as input only for the ogbl-collab dataset. I think it may be useful to improve the performance. Anyway, I will remove it in the next commit.